contracts/zns-pausable/registry/ZNSRegistryPausable.sol

```diff
@@ -0,0 +1,353 @@
+     * about each domain
+     */
+    mapping(bytes32 domainHash => DomainRecord domainRecord) internal records;
+
+    /**
+     * @notice Mapping of `owner` => `operator` => `bool` to show accounts that
+     * are or aren't allowed access to domains that `owner` has access to.
+     * Note that operators can NOT change the owner of the domain, but can change
+     * the resolver or resolver records.
+     */
+    mapping(address owner => mapping(address operator => bool isOperator))
+        internal operators;
+
+    bool private _paused;
+
+    /**
+     * @notice Revert if `msg.sender` is not the owner or an operator allowed by the owner
+     * @param domainHash the hash of a domain's name
+     */
+    modifier onlyOwnerOrOperator(bytes32 domainHash) {
+        require(
+            isOwnerOrOperator(domainHash, msg.sender),
+            "ZNSRegistry: Not authorized"
+        );
+        _;
+    }
+
+    /**
+     * @notice Revert if `msg.sender` is not the owner. Used for owner restricted functions.
+     * @param domainHash the hash of a domain's name
+     */
+    modifier onlyOwner(bytes32 domainHash) {
+        require(
+            records[domainHash].owner == msg.sender,
+            "ZNSRegistry: Not the Name Owner"
+        );
+        _;
+    }
+
+    modifier whenNotPaused() {
+        require(!paused(), "ZNSRegistry: Contract is paused");
+        _;
+    }
+
+    modifier whenPaused() {
+        require(paused(), "ZNSRegistry: Contract is not paused");
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Initializer for the `ZNSRegistry` proxy.
+     * @param accessController_ The address of the `ZNSAccessController` contract
+     * @dev ! The owner of the 0x0 hash should be a multisig ideally, but EOA can be used to deploy !
+     * > Admin account deploying the contract will be the owner of the 0x0 hash !
+     */
+    function initialize(address accessController_) external override initializer {
+        records[0x0].owner = msg.sender;
+        _setAccessController(accessController_);
+    }
+
+    /**
+     * @notice Checks if a given domain exists
+     * @param domainHash The hash of a domain's name
+     */
+    function exists(bytes32 domainHash) external view override returns (bool) {
+        return _exists(domainHash);
+    }
+
+    /**
+     * @notice Checks if provided address is an owner or an operator of the provided domain
+     * @param domainHash The hash of a domain's name
+     * @param candidate The address for which we are checking access
+     */
+    function isOwnerOrOperator(
+        bytes32 domainHash,
+        address candidate
+    ) public view override returns (bool) {
+        address owner = records[domainHash].owner;
+        return candidate == owner || operators[owner][candidate];
+    }
+
+    function paused() public view returns (bool) {
```

Comment:
This should be grouped with the other functions from pausable to be consistent with the other contracts. 

Also, no `virtual` on this one but it is present in other places?